pkg/svcat/service-catalog/broker.go

```diff
@@ -166,26 +176,55 @@ func (sdk *SDK) Register(brokerName string, url string, opts *RegisterOptions) (
 	return result, nil
 }
 
-// Sync or relist a broker to refresh its catalog metadata.
-func (sdk *SDK) Sync(name string, retries int) error {
-	for j := 0; j < retries; j++ {
-		catalog, err := sdk.RetrieveBroker(name)
-		if err != nil {
-			return err
+// Sync or relist a broker to refresh its broker metadata.
+func (sdk *SDK) Sync(name string, scopeOpts ScopeOptions, retries int) error {
+	success := false
+
+	for j := 0; j < retries && !success; j++ {
+
+		if scopeOpts.Scope.Matches(NamespaceScope) {
+			namespace := scopeOpts.Namespace
+			broker, err := sdk.RetrieveNamespacedBroker(namespace, name)
+			if err == nil {
+				broker.Spec.RelistRequests = broker.Spec.RelistRequests + 1
+
+				_, err = sdk.ServiceCatalog().ServiceBrokers(namespace).Update(broker)
+				if err == nil {
+					success = true
+					fmt.Printf("synced servicebroker (%s) in name space (%s)\n", name, scopeOpts.Namespace)
+				}
+				if err != nil && !apierrors.IsConflict(err) {
+					return fmt.Errorf("could not sync service broker (%s)", err)
+				}
+			}
 		}
 
-		catalog.Spec.RelistRequests = catalog.Spec.RelistRequests + 1
+		if scopeOpts.Scope.Matches(ClusterScope) {
+			broker, err := sdk.RetrieveBroker(name)
+			if err == nil {
+				broker.Spec.RelistRequests = broker.Spec.RelistRequests + 1
 
-		_, err = sdk.ServiceCatalog().ClusterServiceBrokers().Update(catalog)
-		if err == nil {
-			return nil
+				_, err = sdk.ServiceCatalog().ClusterServiceBrokers().Update(broker)
+				if err == nil {
+					success = true
+					fmt.Printf("synced clusterservicebroker (%s)\n", name)
+				}
+				if err != nil && !apierrors.IsConflict(err) {
+					return fmt.Errorf("could not sync service broker (%s)", err)
+				}
+			}
 		}
-		if !apierrors.IsConflict(err) {
-			return fmt.Errorf("could not sync service broker (%s)", err)
+		// success to update, no need to retry again
+		if success {
+			break
 		}
 	}
 
-	return fmt.Errorf("could not sync service broker after %d tries", retries)
+	if !success {
+		return fmt.Errorf("could not sync service broker %s", name)
```

Comment:
If the broker isn't found, we give a generic error with no reason why sync failed.  Can you add the err to this?  ie 
```go
return fmt.Errorf("could not sync service broker %s: %s", name, err)
```

Reply:
Maybe I'm reading the function above wrong but when we do this

```
err != nil && !apierrors.IsConflict(err)
```
We bail immediately if it's any real error (other than a 409). So it only retries 3 times in case of repeated conflicts. 

Not saying printing the error is bad, just clarifying that for example if the broker isn't found, we do print the error message.